automation_use_cases/scan_automation/lib/helpers/scan_automation.py

```diff
@@ -0,0 +1,136 @@
+
+        # Step 1 - Read values from config file
+        scan_pairs_names, interval = read_settings(settings)
+
+        # Step 2 - Grab IDs for application names and scan config names
+        scan_pairs_ids = get_ids(api, scan_pairs_names)
+        
+        # Step 3 - Submit all scans
+        scan_ids = submit_scans(api, scan_pairs_ids)
+        id_to_names = {} # scan ID - > (app name, scan config name)
+        for i in range(0, len(scan_ids)):
+            scan_id = scan_ids[i]
+            id_to_names[scan_id] = scan_pairs_names[i]
+
+        # Step 4 - Track scan status
+        track_scans(api, scan_ids, id_to_names, interval)
+
+        # Step 5 - Report Findings
+        report_findings(api, scan_ids, id_to_names)
+
+    except Exception as e:
+        logging.error(f"Encountered error while creating scans: {e}")
+
+def report_findings(api: InsightAppSec, scan_ids: [str], id_to_names: dict):
+    """
+    Given list of scan IDs, report on number of vulnerabilities found
+    """
+    logging.info("REPORTING VULNERABILITY COUNTS OF SCANS... (Scan ID, App Name, Scan Config Name): NUM VULNS")
+    for scan_id in scan_ids:
+        details = api.search('VULNERABILITY', f"vulnerability.scans.id='{scan_id}'")
+        num_findings = details.get("metadata").get("total_data")
+        logging.info(f"({scan_id}, {id_to_names.get(scan_id)[0]}, {id_to_names.get(scan_id)[1]}: {num_findings}")
+
+
+def track_scans(api: InsightAppSec, scan_ids: [str], id_to_names: dict, interval: int):
+    """
+    Tracks the status of all current scans being run
+    """
+    scan_ids_copy = list(scan_ids)
+    stop_criteria = ["COMPLETE", "FAILED"]
+    start = time.time()
+    logging.info("CHECKING STATUS OF REMAINING SCANS... (Scan ID, App Name, Scan Config Name): STATUS")
+    for scan_id in scan_ids_copy:
+        log_status(api, scan_id, id_to_names)
+    while len(scan_ids_copy):
+        if time.time() - start > interval:
+            logging.info("CHECKING STATUS OF REMAINING SCANS... (Scan ID, App Name, Scan Config Name): STATUS")
+            to_remove = []
+            for scan_id in scan_ids_copy:
+                scan_status = log_status(api, scan_id, id_to_names)
+                if scan_status in stop_criteria:
+                    to_remove.append(scan_id)
+            for scan_id in to_remove: # remove scans that completed or failed
+                scan_ids_copy.remove(scan_id)
+            start = time.time()
+
+def read_settings(settings: dict):
+    """
+    Takes in settings dictionary and returns list of tuples (app name, scan config name)
+    """
+    scan_pairs = [] # list of tuples (app name, scan config name)
+    for scan in settings.get("scan_info"):
+        app_name = scan.get("app_name")
+        scan_config_name = scan.get("scan_config_name")
+        if app_name is None or scan_config_name is None:
+            logging.error("Application name and scan config must be set up in configuration file")
+            return
```

Comment:
Going contrary to this, since scans can take a long time it might be nice to get immediate feedback that one of the configurations isn't complete?  Just a thought.

Reply:
We'd need more specific messaging in that case. Something stating more directly that there is a blank app or scan config in the settings file that must be updated for the integration to successfully run.